app/src/test/java/com/swent/suddenbump/model/user/UserRepositoryFirestoreTest.kt

```diff
@@ -2241,6 +2262,124 @@ class UserRepositoryFirestoreTest {
+                  "phoneNumber" to user.phoneNumber,
+                  "emailAddress" to user.emailAddress))
+
+      // Answering the call to addOnSuccessListener and simulating success with the mock data
+      doAnswer { invocation ->
+            val listener = invocation.arguments[0] as OnSuccessListener<DocumentSnapshot>
+            listener.onSuccess(
+                mockUserDocumentSnapshot) // Call onSuccess with the mock user document snapshot
+            mockTask // Return the mock task
+          }
+          .`when`(mockTask)
+          .addOnSuccessListener(any()) // Mocking the success listener being added for this task
+
+      // Mocking the downloadImage function in the image repository
+      doAnswer { invocation ->
+            val pathArgument = invocation.getArgument<String>(0)
+            val onSuccess = invocation.getArgument<(ImageBitmap) -> Unit>(1)
+            onSuccess(profilePicture) // Simulate the image download success, providing a mocked
+            // ImageBitmap
+            null // No return needed
+          }
+          .`when`(mockImageRepository)
+          .downloadImage(eq(profilePicturePath), any(), any())
+
+      // Mocking Tasks.whenAllSuccess to handle multiple tasks
+      `when`(Tasks.whenAllSuccess<DocumentSnapshot>(any<List<Task<DocumentSnapshot>>>()))
+          .thenReturn(mockTaskList)
+
+      // Answering the call to addOnSuccessListener on the list of tasks (for friends)
+      doAnswer { invocation ->
+            val listener = invocation.arguments[0] as OnSuccessListener<List<DocumentSnapshot>>
+            listener.onSuccess(
+                listOf(mockedFriendsDocumentSnapshot)) // Simulate success with a list of friend
+            // documents
+            mockTaskList // Return the mock task list
+          }
+          .`when`(mockTaskList)
+          .addOnSuccessListener(any()) // Mock the success listener for this task list
+
+      // Simulate no failure listener being called for the mockTaskList
+      doAnswer { mockTaskList }.`when`(mockTaskList).addOnFailureListener(any())
+
+      // Access the private field for imageRepository in the UserRepositoryFirestore class
+      val imageRepositoryField =
+          UserRepositoryFirestore::class.java.getDeclaredField("imageRepository")
+      imageRepositoryField.isAccessible = true // Make the field accessible
+      imageRepositoryField.set(
+          userRepositoryFirestore, mockImageRepository) // Inject the mock image repository
+
+      // Call the method under test: getUserFriends
+      userRepositoryFirestore.getUserFriends(
+          uid = user.uid,
+          onSuccess = {}, // Simulate success callback
+          onFailure = {
+            fail("Failure callback should not be called")
+          }) // Fail if the failure callback is called
+
+      // Verify that the image repository's downloadImage function was called
+      verify(mockImageRepository).downloadImage(any(), any(), any())
+    }
+  }
+
```

Comment:
Amazing job with this test! The mocking must have been quite challenging but I see you handled this quite well :)